napari/_qt/layer_controls/qt_image_controls_base.py

```diff
@@ -142,15 +145,28 @@ def changeColor(self, text):
     def _on_contrast_limits_change(self):
         """Receive layer model contrast limits change event and update slider."""
         with qt_signals_blocked(self.contrastLimitsSlider):
+            self.contrastLimitsSlider.setValue(self.layer.contrast_limits)
+
+        if self.clim_popup:
+            with qt_signals_blocked(self.clim_popup.slider):
+                self.clim_popup.slider.setValue(self.layer.contrast_limits)
+
+    def _on_contrast_limits_range_change(self):
+        """Receive layer model contrast limits change event and update slider."""
+        with qt_signals_blocked(self.contrastLimitsSlider):
+            decimals = range_to_decimals(
+                self.layer.contrast_limits_range, self.layer.dtype
+            )
             self.contrastLimitsSlider.setRange(
                 *self.layer.contrast_limits_range
             )
-            self.contrastLimitsSlider.setValue(self.layer.contrast_limits)
+            self.contrastLimitsSlider.setSingleStep(10**-decimals)
```

Comment:
This was only happening on init, which messed up the slider if the `range` changed a lot.